sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CountIf.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions.aggregate
+
+import org.apache.spark.sql.catalyst.analysis.TypeCheckResult
+import org.apache.spark.sql.catalyst.expressions.{Expression, ExpressionDescription, ImplicitCastInputTypes, UnevaluableAggregate}
+import org.apache.spark.sql.types.{AbstractDataType, BooleanType, DataType, LongType}
+
+@ExpressionDescription(
+  usage = """
+    _FUNC_(expr) - Returns the number of `TRUE` values for the expression.
+      This function is equivalent to `count(CASE WHEN expr THEN 1 END)`.
```

Comment:
Initially, I recommended this to give a hint to the users like the other SQL engines. The reason why I chose this expression instead of `Count(NullIf(...))` which is used in this PR with `RuntimeReplaceable` is that `Count(NullIf(...))` doesn't work like new `count_if` due to the type casting.

For the following case, `Count(NullIf(...))` works while `count_if` doesn't.
```sql
spark-sql> select count(nullif(a, false)) from values (1) T(a);
1

spark-sql> select count_if(a) from values (1) T(a);
Error in query: cannot resolve 'count_if(T.a)' due to data type mismatch: function count_if requires boolean type,

spark-sql> select count(case when a then 1 end) from values (1) T(a);
Error in query: cannot resolve 'CASE WHEN T.`a` THEN 1 END' due to data type mismatch: WHEN expressions in CaseWhen
```

In short, new `count_if`'s behavior is the same with `count(CASE WHEN expr THEN 1 END)`. However, while reviewing this PR again, I notice that this might mislead the **developers** because we are using `count(nullif(...))` technically.

To sum up, we cannot give the simple fallback example here. Both ones are inadequate. We had better remove this line. So, could you remove this line again, @cryeo ? Sorry, it's my bad.

Reply:
Okay, thanks for your advice.